publisher-sdk/src/main/java/com/criteo/publisher/logging/PublisherCodeSanitizer.kt

```diff
@@ -0,0 +1,220 @@
+/*
+ *    Copyright 2020 Criteo
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.criteo.publisher.logging
+
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.ObjectInputStream
+import java.io.ObjectOutputStream
+import java.lang.reflect.Field
+
+class PublisherCodeSanitizer {
+
+  private val criteoPackagePrefix = "com.criteo."
+  private val allowedPackagePrefixes = listOf(
+      // Java
+      "java.",
+      "javax.",
+      "sun.",
+      "com.sun.",
+
+      // Kotlin
+      "com.intellij.",
+      "org.jetbrains.",
+
+      // Android
+      "android.",
+      "com.android.",
+      "androidx.",
+      "dalvik.",
+
+      // Libs
+      "com.google",
+      "com.mopub",
+      "org.json",
+      "com.squareup.",
+      "org.junit."
```

Comment:
We're not using `com.uber.*` lib. Any reason why we would need it ? Do they generate contexts that could be useful ? (Note that with Android+JDK we already know if we're in a service, activity, application, thread pool, ...)

Here, Google and MoPub are selected because of mediations adapters. Then JSON and Square because they are directly used by the SDK. Finally JUnit is there only to accomodate tests.

Reply:
Ok I misunderstood what you were trying to achieve here. You should document that then.

Reply:
Done